enumerable_methods.rb

```diff
@@ -0,0 +1,111 @@
+# This is the enumerable methods project done by Mohammad and Kingsley
+module Enumerable
+  def my_each
+    i = 0
+    object = self
+    while i < object.length
+      yield object[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    object = self
+    while i < object.length
+      yield(object[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    object = self
+    object.my_each do |x|
+      (new_arr << x) if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all?
+    object = self
+    object.my_each do |x|
+      return false if yield(x) == false
+    end
+    true
+  end
+
+  def my_any?
+    object = self
+    object.my_each do |x|
+      return true if yield(x) == true
```

Comment:
This could be simplified into `return yield(x)`

Reply:
Hi @talihomz 
`return yield x` will always return `false` unless only the first element match the condition, In this case, I need to loop through all the elements until the end. right?